### ISSUE TITLE
Boston Housing Example issue

### ISSUE THREAD

**Issue by waynefoltaERI** [statsmaths/kerasR#21]
Super Package!

The very first example in the great tutorial, Boston Housing, seems to have an error in the R code that causes the bad performance. The problems are: 1) the X_train and X_test data are separately scaled in R, and 2) the X_test data is also normalized in the prediction. The first is just a nit-pick of mine and doesn't matter that much, but the second seems to really make a difference. I believe this is remedied with:

```
mod <- Sequential ()

mod$add (Dense (units=50, input_shape=13))
mod$add (Activation ("relu"))

mod$add (Dense (units=1))

keras_compile (mod,  loss="mse", optimizer=RMSprop())

boston  <- load_boston_housing ()
X_train <- scale (boston$X_train)
Y_train <- boston$Y_train
X_test  <- scale (boston$X_test, center=attr (X_train, "scaled:center"), scale=attr (X_train, "scaled:scale"))
Y_test  <- boston$Y_test

keras_fit (mod, X_train, Y_train, batch_size=32, epochs=200, verbose=1, validation_split=0.1)

pred <- keras_predict (mod, X_test)
sd (as.numeric (pred) - Y_test) / sd (Y_test)

plot (Y_test, pred)
abline (0, 1)

```

**Comment by statsmaths** [comment]
Thanks for the suggestion. You are absolutely right that we should not be double normalising the training data. I've pushed an update in 146f797b8e132dc2f94a3b730f8e7b2b49d8f36a that only uses the `normalize` rather than R's scale function.

As for normalising the training and testing sets separately, I agree that in many ways it is a more fair comparison to use the scaling factors learned from the training set and your code for doing this is exactly correct. However, in image processing (the most prominent application for Keras), it is generally more common to separately scale each input. In fact, often each row of the dataset is separately scaled to have unit norm.
